src/pulselistener/pulselistener/mercurial.py

```diff
@@ -115,12 +115,30 @@ def clean(self):
         base, patches = self.phabricator_api.load_patches_stack(self.repo, diff, default_revision='central')
         assert len(patches) > 0, 'No patches to apply'
 
+        # Load all the diffs details with commits messages
+        diffs = self.phabricator_api.search_diffs(
+            diff_phid=[p[0] for p in patches],
+            attachments={
+                'commits': True,
+            }
+        )
+        commits = {
+            diff['phid']: diff['attachments']['commits'].get('commits', [])
+            for diff in diffs
+        }
+
         # Apply the patches and commit them one by one
         for diff_phid, patch in patches:
-            logger.info('Applying patch', phid=diff_phid)
+            commit = commits.get(diff_phid)
+            if commit:
+                message = '{}\nDiff: {}'.format(commit[0]['message'], diff_phid)
```

Comment:
The Diff ID is not already in the commit message?

Reply:
Maybe you can keep the same convention as commits landed via Phab, which add `Differential Revision: https://phabricator.services.mozilla.com/DXXX` at the end (e.g. https://hg.mozilla.org/mozilla-central/rev/788b5dbea4e8).

Reply:
Only the revision `D1234` is mentionned, i'll use the format you mentionned for PHID, but there are no url pointing to a `PHID-DIFF`